lib/rucio/api/replica.py

```diff
@@ -204,15 +204,10 @@ def list_replicas(dids, schemes=None, unavailable=False, request_id=None,
 
     for rep in replicas:
         # 'rses' and 'states' use rse_id as the key. This needs updating to be rse.
-        keys = ['rses', 'states']
-        for k in keys:
-            old_dict = rep.get(k, None)
+        for k in ('rses', 'states'):
+            old_dict = rep.get(k)
             if old_dict is not None:
-                new_dict = {}
-                for rse_id in old_dict:
-                    rse = get_rse_name(rse_id=rse_id, session=session) if rse_id is not None else None
-                    new_dict[rse] = old_dict[rse_id]
-                rep[k] = new_dict
+                rep[k] = rep.pop(k + "_by_name", None)
```

Comment:
This pattern: using the result dictionary as a storage for temporary data which is then removed from the dict is extremely confusing and prone to introduction of bugs in the future. I just spent many months getting rid of this behavior in conveyor and fixing numerous bugs linked to it.

I would rather pass an additional function parameter to core.replica.list_replicas(). For example: `group_rses_by_name = True/False` and directly return the data in the correct format. My IMHO...

Reply:
Thanks Radu. Done.